_posts/2015-09-11-android-picker-knob.md

```diff
@@ -0,0 +1,222 @@
+---
+layout: post
+title:  "Android Picker Knob"
+date:   2015-09-11 11:11:11
+author: anuj
+categories: Technical Android
+---
+
+In a game we are building, we needed a custom widget that looks like the volume control knob of old.
+
+![Picker knob][screenshot]
+
+We went ahead and built a custom view for this. The view itself is available as a library [here][github-page]. Below is a brief description of the creation process.
+
+#### Custom or Compound view?
+Android provides a wide array of widgets that can be combined to create most UI elements. It is always a good idea to reuse existing widgets in a [Compound View][compound-view] if possible. For the picker knob, existing widgets were not ideal. Custom view it was then.
+
+#### Strategy
+
+__Mathematics__
+To get realistic physics and behavior, we visualized a strip of equidistant markers wound on a circular dial. Based on the current rotation, the view identifies the markers on the front half of the dial, projects them on a 1-d space (the lins AB in the diagram below) and draws them on the screen.
```

Comment:
wound as in past tense of wind (to coil or twine about something)

lins is a type for line